### DISCUSSION TITLE
$queryString should support models

### DISCUSSION THREAD

**Discussion by dominiced** [livewire/livewire#1888] [3. Feature ideas]
I really like the $queryString functionality for keeping the URL up to date when changing between pages etc.

On my website I've got a table listing all users, and when I click a row in that table a form appears, the form is a livewire component and it's bound to the user model which was clicked on in the table. I've also created the route such that you can go to /users/{id?} and it'll open up the form for that user as if you clicked on it in the table. However, that's only if you follow a pre-created url for it. I'd love to have a feature where opening and closing the livewire modal form modifies the URL with the user id slug like in my example.

I tried, to see if this is currently supported, but putting the $user property into the $queryString array caused the entire model to be outputted to the query string.

What do you think? Thanks for reading.

**Comment by iksaku** [comment]
This idea sounds to me a lot like [Inertia's Partial Reloads](https://inertiajs.com/partial-reloads), which doesn't sound bad at all!

Is that Inertia Feature close enough to your idea? If so, we can start discussing the possible implementation behind it.

If not exactly what you meant, could you try to provide an extended explanation?

**Comment by dominiced** [comment]
I think I'm referring to something slightly different, but I read the description of partial reloads and I like your thinking, partial reloads would be a great addition too!

What I'm after is quite simple to describe, all I'd like is to be able to use a model property in a query string rather than the model itself.

Let's say we have a public property $user in our component, and it's a User model. We might be binding to the name and email address inside our view component thanks to Livewire allowing this... https://laravel-livewire.com/docs/2.x/properties#binding-models

However, if you then add that user property into the $queryString property like so...
```
protected $queryString = [
        'user',
];
```

Then it throws the entire model into the query string is ugly. What would be better is to have a $queryString property like so...
```
protected $queryString = [
        'user.id',
];
```
This way you're referencing an underlying property to act as the query string parameter, and you're referencing it much the same way you are in the view when binding. However, Livewire currently throws an exception for the following syntax because it can't find a property 'user.id' as it doesn't exist of course.
